client-api/src/main/java/org/sonarsource/sonarlint/core/client/api/util/FileUtils.java

```diff
@@ -152,4 +162,47 @@ public static void replaceDir(Consumer<Path> dirContentUpdater, Path target, Pat
     FileUtils.mkdirs(target.getParent());
     FileUtils.moveDir(work, target);
   }
+  
+  /**
+   * On Windows, retries the provided IO operation a number of times, in an effort to make the operation succeed.
+   * 
+   * Operations that might fail on Windows are file & directory move, as well as file deletion. These failures 
+   * are typically caused by the virus scanner and/or the Windows Indexing Service. These services tend to open a file handle 
+   * on newly created files in an effort to scan their content.
+   * 
+   * @param runnable the runnable whose execution should be retried
+   */
+  private static void retry(IORunnable runnable) throws IOException {
+    for (int retries = isWindows()? 30: 1; retries > 0; retries--) {
+      if (retries > 1) {    	  
```

Comment:
Instead of checking the current retry count in every iteration of the loop, I would use a flag, set after successful run, and then after the loop, run one last time if the flag was not set.

Reply:
That was how the code looked like originally. A question of preference, if you ask me. But since I'm anyway fixing stuff, I can fix that as well. Anything against early exits/returns?